packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -90,35 +99,168 @@ class BottomNavigationBar extends StatefulWidget {
   /// The index into [items] of the current active item.
   final int currentIndex;
 
-  /// Defines the layout and behavior of a [BottomNavigationBar].
+  /// Defines the layout and behavior of a [BottomNavigationBar].  See
+  /// documentation for [BottomNavigationBarType] for information on the meaning
+  /// of different types.
   final BottomNavigationBarType type;
 
   /// The color of the selected item when bottom navigation bar is
-  /// [BottomNavigationBarType.fixed].
+  /// [BottomNavigationBarType.fixed].  [fixedColor] must be null if the [type]
```

Comment:
ditto.

also to avoid starting a sentence with a lowercase letter use "`The [fixedColor] argument...`".

Reply:
Also you say it must be null, but that doesn't seem to be enforced.

Reply:
Doesn't `assert(type == BottomNavigationBarType.fixed || fixedColor == null)` assert that?  If the type is not fixed, then if the value is something other than null the assert will fire.

However, I think it would be less surprising to have fixedColor just be ignored if the `type` is not `fixed`, so that's what I changed it to.  Let me know if that was somehow intentional in the original.

Reply:
I meant in the constructor (but maybe I missed that?). Anyway, either way is fine.